account-integrations/safe/src/SafeECDSARecoveryPlugin.sol

```diff
@@ -71,25 +70,24 @@ contract SafeECDSARecoveryPlugin {
 
     function addRecoveryAccount(
         bytes32 recoveryHash,
-        address safe,
+        address owner,
         address ecsdaPlugin
     ) external {
-        if (safe == address(0)) revert SAFE_ZERO_ADDRESS();
+        address safe = msg.sender;
```

Comment:
@jzaki do you think we should change this to assume it can be called elsewhere beyond a safe, e.g. directly by the owner?

To do that we would need to change how we use `msg.sender`, as it would be different in each case. If calling directly, we couldn't assume `msg.sender` is the safe. Maybe we could do something like the following in `addRecoveryAccountWithoutMsgSender`:

```solidity
    // How it's currently done
    function addRecoveryAccount(
        bytes32 recoveryHash,
        address owner,
        address ecsdaPlugin
    ) external {
        address safe = msg.sender;

        bool moduleEnabled = ISafe(safe).isModuleEnabled(address(this));
        if (!moduleEnabled) revert MODULE_NOT_ENABLED();

        address expectedOwner = ISafeECDSAPlugin(ecsdaPlugin).getOwner(safe);
        if (owner != expectedOwner) revert INVALID_OWNER(expectedOwner, owner);

        ecdsaRecoveryStorage[safe] = ECDSARecoveryStorage(recoveryHash);
    }

    // How we could change it to not rely on msg.sender (pseudocode)
    function addRecoveryAccountWithoutMsgSender(
        bytes32 recoveryHash,
        bytes memory ownerSignature,
        address safe,
        address ecsdaPlugin
    ) external {
        bool moduleEnabled = ISafe(safe).isModuleEnabled(address(this));
        if (!moduleEnabled) revert MODULE_NOT_ENABLED();

        bytes32 ownersSafeHash = keccak256(abi.encodePacked(safe));
        bytes32 ethSignedMessage = ownersSafeHash.toEthSignedMessageHash();
        address owner = ethSignedMessage.recover(ownerSignature);

        address expectedOwner = ISafeECDSAPlugin(ecsdaPlugin).getOwner(safe);
        if (owner != expectedOwner) revert INVALID_OWNER(expectedOwner, owner);

        ecdsaRecoveryStorage[safe] = ECDSARecoveryStorage(recoveryHash);
    }
```

maybe another option might be to create functions for each scenario like `addRecoveryAccountFromSafe` and `addRecoveryAccountFromOwner`. Feels better to use one function though.

Curious to get your thoughts